### ISSUE TITLE
Style/MutableConstant false positive with frozen_string_literal: true and line continuation

### ISSUE THREAD

**Issue by JacobEvelyn** [rubocop/rubocop#10134]
When `# frozen_string_literal: true` is set, a constant string that includes a line continuation (`\`) is incorrectly flagged by the `Style/MutableConstant` cop, even though the constant is indeed frozen.

--------

## Expected behavior

RuboCop should not flag this as a `Style/MutableConstant` violation.

## Actual behavior

RuboCop wants a `.freeze` at the end of the string literal to appease the `Style/MutableConstant` cop.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

C = 'this is a string '\
    'wrapped onto another line'
```

## RuboCop version

```
$ bundle exec rubocop -V
1.21.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 3.0.2 x86_64-darwin20)
  - rubocop-performance 1.11.5
  - rubocop-rails 2.12.2
  - rubocop-rspec 2.4.0
```
